File: LLM-Translations/GPT-4/avatar/Python/C/codeforces_350_B.c
#include <stdio.h>
#define MAX 1000

int main() {
    int n;
    scanf("%d", &n);
    int t[MAX], a[MAX], cnt[MAX], ans[MAX], crt[MAX];
    t[0] = 0;
    a[0] = 0;
    for(int i = 1; i <= n; i++) {
        scanf("%d", &t[i]);
    }
    for(int i = 1; i <= n; i++) {
        scanf("%d", &a[i]);
    }
    for(int i = 0; i <= n; i++) {
        cnt[i] = 0;
    }
    for(int i = 1; i <= n; i++) {
        cnt[a[i]]++;
    }
    int ans_len = 0;
    for(int i = 1; i <= n; i++) {
        if(t[i] == 1) {
            int crt_len = 1;
            crt[0] = i;
            int x = a[i];
            while(cnt[x] == 1) {
                crt[crt_len++] = x;
                x = a[x];
            }
            if(crt_len > ans_len) {
                ans_len = crt_len;
                for(int j = 0; j < crt_len; j++) {
                    ans[j] = crt[j];
                }
            }
        }
    }
    for(int i = 0; i < ans_len / 2; i++) {
        int temp = ans[i];
        ans[i] = ans[ans_len - i - 1];
        ans[ans_len - i - 1] = temp;
    }
    printf("%d\n", ans_len);
    for(int i = 0; i < ans_len; i++) {
        printf("%d ", ans[i]);
    }
    return 0;
} // End of Code.

File: LLM-Translations/GPT-4/avatar/Python/Go/codeforces_350_B.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func main() {
	var n int
	_, _ = fmt.Scan(&n)

	t := make([]int, n+1)
	tStr := ""
	_, _ = fmt.Scanln(&tStr)
	tStr = strings.TrimSuffix(tStr, "\n")
	tStrArr := strings.Split(tStr, " ")
	for i, v := range tStrArr {
		t[i+1], _ = strconv.Atoi(v)
	}

	a := make([]int, n+1)
	aStr := ""
	_, _ = fmt.Scanln(&aStr)
	aStr = strings.TrimSuffix(aStr, "\n")
	aStrArr := strings.Split(aStr, " ")
	for i, v := range aStrArr {
		a[i+1], _ = strconv.Atoi(v)
	}

	ans, cnt := make([]int, 0), make([]int, n+1)
	for _, v := range a {
		cnt[v]++
	}
	for i := 1; i <= n; i++ {
		if t[i] == 1 {
			crt := []int{i}
			x := a[i]
			for cnt[x] == 1 {
				crt = append(crt, x)
				x = a[x]
			}
			if len(crt) > len(ans) {
				ans = crt[:]
			}
		}
	}

	for i := 0; i < len(ans)/2; i++ {
		j := len(ans) - i - 1
		ans[i], ans[j] = ans[j], ans[i]
	}

	fmt.Println(len(ans))
	fmt.Println(strings.Trim(strings.Replace(fmt.Sprint(ans), " ", " ", -1), "[]"))
}

// End of Code

